src/game/logic/object/locomotor.h

```diff
@@ -0,0 +1,154 @@
+/**
+ * @file LocomotorStore
+ *
+ * @author Duncans_pumpkin
+ *
+ * @brief LocomotorStore
+ *
+ * @copyright Thyme is free software: you can redistribute it and/or
+ *            modify it under the terms of the GNU General Public License
+ *            as published by the Free Software Foundation, either version
+ *            2 of the License, or (at your option) any later version.
+ *            A full copy of the GNU General Public License can be found in
+ *            LICENSE
+ */
+#include "mempoolobj.h"
+#include "namekeygenerator.h"
+#include "overridable.h"
+#include "subsysteminterface.h"
+#include <map>
+
+class INI;
+
+class Locomotor;
+
+enum LocomotorBehaviorZ
+{
+    Z_AXIS_BEHAVIOUR_NO_Z_MOTIVE_FORCE,
+    Z_AXIS_BEHAVIOUR_SEA_LEVEL,
+    Z_AXIS_BEHAVIOUR_SURFACE_RELATIVE_HEIGHT,
+    Z_AXIS_BEHAVIOUR_ABSOLUTE_HEIGHT,
+    Z_AXIS_BEHAVIOUR_FIXED_SURFACE_RELATIVE_HEIGHT,
+    Z_AXIS_BEHAVIOUR_FIXED_ABSOLUTE_HEIGHT,
+    Z_AXIS_BEHAVIOUR_FIXED_RELATIVE_TO_GROUND_AND_BUILDINGS,
+    Z_AXIS_BEHAVIOUR_RELATIVE_TO_HIGHEST_LAYER,
+};
+
+enum LocomotorAppearance
+{
+    LOCOMOTOR_APPEARANCE_TWO_LEGS,
+    LOCOMOTOR_APPEARANCE_FOUR_WHEELS,
+    LOCOMOTOR_APPEARANCE_TREADS,
+    LOCOMOTOR_APPEARANCE_HOVER,
+    LOCOMOTOR_APPEARANCE_THRUST,
+    LOCOMOTOR_APPEARANCE_WINGS,
+    LOCOMOTOR_APPEARANCE_CLIMBER,
+    LOCOMOTOR_APPEARANCE_OTHER,
+    LOCOMOTOR_APPEARANCE_MOTORCYCLE,
+};
+
+class LocomotorTemplate : public Overridable
+{
+    IMPLEMENT_POOL(LocomotorTemplate);
+
+public:
+    virtual ~LocomotorTemplate() override{};
+    LocomotorTemplate();
+
+    void Set_Name(const Utf8String &newName) { m_name = newName; }
+    void Validate() const;
+
+    static FieldParse *Get_Field_Parse() { return s_fieldParseTable; }
+
+private:
+    Utf8String m_name;
+    int32_t m_surfaces;
```

Comment:
These are bit flags. Check see what convention is for bitflags

Reply:
char *_TheLocomotorSurfaceTypeNames[] =
{ "GROUND", "WATER", "CLIFF", "AIR", "RUBBLE"};
looks like name for the enum was LocomotorSurfaceType, not sure if there's a entry 0 or does it start with ground tho....

Reply:
Convention seems to be leave as `int32_t`